spec/rubocop/cop/style/class_structure_spec.rb

```diff
@@ -0,0 +1,167 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::ClassStructure, :config do
+  subject(:cop) { described_class.new(config) }
+  let(:config) do
+    RuboCop::Config.new('Style/ClassStructure' => {
+      'ExpectedOrder' => [
+        'extend',
+        'include',
+        'inner_class',
+        'constant',
+        'attribute_macro',
+        'macro',
+        'public_class_method',
+        'initialize',
+        'instance_method',
+        'protected_method',
+        'private_method'
+      ],
+      'Categories' => {
+        'macro' => %w[validates validate],
+        'include' => %w[prepend],
+        'attribute_macro' => %w[attr_accessor attr_reader attr_writer]
+      }
+    })
+  end
+
+  context 'with a complete ordered example' do
+    it 'does not create offense' do
+      expect_no_offenses <<-RUBY
+        class Person
+          # extend and include go first
+          extend SomeModule
+          include AnotherModule
+
+          # inner classes
+          CustomError = Class.new(StandardError)
+
+          # constants are next
+          SOME_CONSTANT = 20
+
+          # afterwards we have attribute macros
+          attr_reader :name
+
+          # followed by other macros (if any)
+          validates :name
+
+          # public class methods are next in line
+          def self.some_method
+          end
+
+          # initialization goes between class methods and other instance methods
+          def initialize
+          end
+
+          # followed by other public instance methods
+          def some_method
+          end
+
+          # protected and private methods are grouped near the end
+          protected
+
+          def some_protected_method
+          end
+
+          private
+
+          def some_private_method
+          end
+        end
+      RUBY
+    end
+  end
+  context 'simple example' do
+    let(:code) { <<-RUBY }
+      class Person
+        CONST = 'wrong place'
+        include AnotherModule
+        ^^^^^^^^^^^^^^^^^^^^^ `include` is supposed to appear before `constant` in the class definition.
+        extend SomeModule
+        ^^^^^^^^^^^^^^^^^ `extend` is supposed to appear before `include` in the class definition.
+      end
+    RUBY
+    it { expect_offense(code) }
+
+    specify do
+      expect(autocorrect_source_with_loop(<<-RUBY.strip_indent))
```

Comment:
Might be a squiggly `~>>` heredoc.

Reply:
I was thinking about it but RuboCop is compatible with old versions of Ruby and heredoc was introduced in `2.3`.

I think this cop also can be useful for older versions. Other cops are still using this `<<-` combined with `.strip_indent`.